test/end-to-end/passwordless.test.js

```diff
@@ -1408,6 +1427,7 @@ export function getPasswordlessTestCases({ authRecipe, logId }) {
                     "ST_LOGS SESSION OVERRIDE ADD_FETCH_INTERCEPTORS_AND_RETURN_MODIFIED_FETCH",
                     "ST_LOGS SESSION OVERRIDE ADD_AXIOS_INTERCEPTORS",
                     `ST_LOGS ${logId} OVERRIDE GET_LOGIN_ATTEMPT_INFO`,
+                    `ST_LOGS ${logId} OVERRIDE GET_LOGIN_ATTEMPT_INFO`,
```

Comment:
Why are these events getting called twice?

Reply:
They are part of the state initialization of the passwordless signinup feature component. They are called twice because the state wasn't updated before the remount (unmounted before the `load` dispatch). 
We could add a separate bit of state with useRef that could avoid this test change, but it doesn't feel appropriate:
- These things are OK to be called multiple times. (For API calls, error and success handlers can be called multiple times as well since they are concerned with the local UI state)
- We'd be adding an extra state to fix behavior that is only an issue during testing.

Reply:
I think you should put it as part of this PR, or on top of this PR.

Reply:
> They are called twice because the state wasn't updated before the remount (unmounted before the load dispatch).

Could you explain this more? Also, will the API calls for consuming the third party code be called twice and the first one is getting aborted?

Reply:
Yep:
API calls are behind `onMountAPICall` which only calls the API only once and saves the resulting promise. This then either calls a response or an error handler which can be called multiple times, because:
- These may be interrupted (they are after a potentially long async operation) while making changes to local state.
- They should be only concerned with UI state, so calling them twice shouldn't be a problem (except for the tests)
That's the cause of redirect handlers being called twice.

`GET_LOGIN_ATTEMPT_INFO` is called twice because it's part of initial state loading and I didn't consider it to be an API call (it just reads localstorage), so I didn't put it behind `onMountAPICall`. It's should not cause a problem and it actually sets the state immediately after `getLoginAttemptInfo` returns.